File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch05/C/P05-TanPlot.cpp
// Plots the tangent function using the continued fraction representation
#include "memalloc.h"
#include "elemfunc.h"
#include "graphlib.h"

int main(int argc, wchar_t** argv)
{
   double h, xmin, xmax, *x, *y;
   int i, n;

   xmin = -1.5e0; xmax = 1.5e0;
   h = 0.01e0;
   n = int((xmax-xmin)/h) + 1;

   x = Vector(1,n); y = Vector(1,n);                      // arrays for plots

   for (i=1; i<=n; i++) {
      x[i] = xmin + (i-1)*h;
      y[i] = Tan(x[i]);
   }

   PyGraph w(argc, argv);
   w.GraphInit(600,600);

   w.Plot(x,y,n,"blue",1,0.2,0.9,0.1,0.9,"x","Tan(x)","");

   w.MainLoop();
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch05/Python/P05-TanPlot.py
# Plots the tangent function using the continued fraction representation
from math import *
from elemfunc import *
from graphlib import *

# main

xmin = -1.5e0; xmax = 1.5e0                                 # plotting domain
h = 0.01e0                                                 # argument spacing
n = int((xmax-xmin)/h) + 1                                 # number of points

x = [0]*(n+1); y = [0]*(n+1)                               # arrays for plots

for i in range(1,n+1):
  x[i] = xmin + (i-1)*h
  y[i] = Tan(x[i])

GraphInit(600,600)

Plot(x,y,n,"blue",1,0.2,0.9,0.1,0.9,"x","Tan(x)","")

MainLoop()
